### PR TITLE
PWGGA/GammaConv: adaptation of HeavyMeson task to work with AODs 

### PR DESCRIPTION
**Pull request by fjonasALICE**
this commit will be used to check on grid if everything works as intended for AODs

### CODE REVIEW

File: PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_CaloMode_pp.C [hunk 1]
```diff
@@ -182,9 +188,21 @@ void AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_CaloMode_pp(
   } else if( trainConfig == 110)  { // Standard EMCal
     cuts.AddCutHeavyMesonCalo("00000113","1111111047032230000","302010708","0103603900000000","0153503000000000");
     cuts.AddCutHeavyMesonCalo("00000113","1111111047032230000","322010708","0103603900000000","0153503000000000"); // with ITS requirement
+  } else if( trainConfig == 111)  { // Test for EMCal (5 TeV)
+    cuts.AddCutHeavyMesonCalo("00010113","1111111047032230000","302010708","0103603900000000","0153503000000000");
+    cuts.AddCutHeavyMesonCalo("00010113","1111111047032230000","322010708","0103603900000000","0153503000000000"); // with ITS requirement
+
+    // systematic studies (EMCal)
+  } else if( trainConfig = 120)   {
```
Comment (nschmidtALICE) [inline PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_CaloMode_pp.C]: Please fix the equal signs!

Comment (fjonasALICE) [reply PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_CaloMode_pp.C]: done

File: PWGGA/GammaConv/AliConvEventCuts.cxx [hunk 2]
```diff
@@ -3508,7 +3510,8 @@ void AliConvEventCuts::GetXSectionAndNTrials(AliMCEvent *mcEvent, Float_t &XSect
             for(Int_t igen = 0; igen < nGenerators; igen++){
               AliGenEventHeader * eventHeaderGen = mch->GetCocktailHeader(igen) ;
               TString name = eventHeaderGen->GetName();
-              if (name.CompareTo("AliGenPythiaEventHeader") == 0 || name.Contains("Pythia8Jets")){
+              printf("---------->Header Name: %s\n",name.Data());
```
Comment (nschmidtALICE) [inline PWGGA/GammaConv/AliConvEventCuts.cxx]: Please remove the printf.

File: PWGGA/GammaConv/AliConversionMesonCuts.h [hunk 3]
```diff
@@ -113,10 +113,15 @@ class AliConversionMesonCuts : public AliAnalysisCuts {
...
+    Bool_t MesonIsSelectedAODMCPiPlPiMiEta(AliAODMCParticle *fMCMother,TClonesArray *AODMCArray, Int_t &labelNegPion, Int_t &labelPosPion, Int_t &labelNeutPion, Double_t fRapidityShift=0);
     Bool_t MesonIsSelectedMCPiPlPiMiPiZero(TParticle *fMCMother,AliMCEvent *mcEvent, Int_t &labelNegPion, Int_t &labelPosPion, Int_t &labelNeutPion, Double_t fRapidityShift=0);
+    Bool_t MesonIsSelectedAODMCPiPlPiMiPiZero(AliAODMCParticle *fMCMother,TClonesArray *AODMCArray, Int_t &labelNegPion, Int_t &labelPosPion, Int_t &labelNeutPion, Double_t fRapidityShift=0);
     Bool_t MesonIsSelectedMCPiZeroGamma(TParticle *fMCMother, AliMCEvent *mcEvent, Int_t &labelNeutPion, Int_t &labelGamma, Double_t fRapidityShift=0);
+    Bool_t MesonIsSelectedAODMCPiZeroGamma(AliAODMCParticle *fMCMother, TClonesArray *AODMCArray, Int_t &labelNeutPion, Int_t &labelGamma, Double_t fRapidityShift=0);
     Bool_t MesonIsSelectedMCChiC(TParticle *fMCMother,AliMCEvent *mcEvent, Int_t &, Int_t &, Int_t &, Double_t fRapidityShift=0. );
+    Bool_t MesonIsSelectedAODMCChiC(AliAODMCParticle *fMCMother,TClonesArray *AODMCArray, Int_t &, Int_t &, Int_t &, Double_t fRapidityShift=0. );
```
Comment (nschmidtALICE) [inline PWGGA/GammaConv/AliConversionMesonCuts.h]: please increase the classDef of the header

File: PWGGA/GammaConv/AliPrimaryPionCuts.h [hunk 4]
```diff
@@ -80,8 +80,11 @@ class AliPrimaryPionCuts : public AliAnalysisCuts {
 
 		// Cut Selection
     Bool_t PionIsSelectedMC(Int_t labelParticle,AliMCEvent *mcEvent);
-	Bool_t TrackIsSelected(AliESDtrack* lTrack);
-	Bool_t PionIsSelected(AliESDtrack* lTrack);
+    Bool_t PionIsSelectedAODMC(Int_t labelParticle, TClonesArray *AODMCTrackArray);
+    Bool_t TrackIsSelected(AliESDtrack* lTrack);
+    Bool_t TrackIsSelectedAOD(AliAODTrack* lTrack);
+    Bool_t PionIsSelected(AliESDtrack* lTrack);
+    Bool_t PionIsSelectedAOD(AliAODTrack* lTrack);
```
Comment (nschmidtALICE) [inline PWGGA/GammaConv/AliPrimaryPionCuts.h]: please increase the classdef

File: PWGGA/GammaConv/AliPrimaryPionSelector.h [hunk 5]
```diff
@@ -46,6 +46,7 @@ class AliPrimaryPionSelector : public AliAnalysisTaskSE {
    //selected electron arrays
    
    Bool_t ProcessESDs();
+   Bool_t ProcessAODs();
```
Comment (nschmidtALICE) [inline PWGGA/GammaConv/AliPrimaryPionSelector.h]: please increase the classdef

**Comment by nschmidtALICE** [comment]
+1

**Pull request merged**